app.rb

```diff
@@ -0,0 +1,227 @@
+require './classroom'
+require './person'
+require './teacher'
+require './student'
+require './book'
+require './rental'
+
+module Utils
+  def self.print_prompt
+    puts '
+            Please choose an option by entering a number:
+            1 | List all books
+            2 | List all people
+            3 | Create a person
+            4 | Create a book
+            5 | Create a rental
+            6 | List all rentals for a given person id
+            7 | Exit'
+    puts ''
+    puts 'Select an option: '
+  end
+
+  def self.read_name
+    print 'Name: '
+    name = gets.chomp
+    name.empty? ? read_name : name
+  end
+
+  def self.read_age
+    print 'Age: '
+    age = gets.chomp.to_i
+    (1..1000).include?(age) ? age : read_age
+  end
+
+  def self.read_permission
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    %w[Y N].include?(permission.capitalize) ? permission.capitalize : read_permission
+  end
+
+  def self.read_specialization
+    print 'Specialization: '
+    specialization = gets.chomp
+    specialization.empty? ? read_specialization : specialization
+  end
+end
+
+class App
+  attr_accessor :user_input
+
+  def initialize
+    puts 'Welcome to School Library App!'
+    Utils.print_prompt
+    @user_input = gets.chomp
+    @default_classroom = Classroom.new('default-classroom')
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def user_input_valid?(user_input, arr)
+    arr.include?(user_input)
+  end
+
+  def student_info
+    age = Utils.read_age
+    name = Utils.read_name
+    has_parent_permission = Utils.read_permission == 'Y'
+    [age, name, has_parent_permission]
+  end
+
+  def teacher_info
+    age = Utils.read_age
+    name = Utils.read_name
+    specialization = Utils.read_specialization
+    [age, name, specialization]
+  end
+
+  def create_person
+    print "\nDo you want to create a student (1) or a teacher (2)? [Input the number]: "
+    @user_input = gets.chomp
+    create_person unless user_input_valid?(user_input, %w[1 2])
+
+    if @user_input == '1'
+      age, name, has_parent_permission = student_info
+      person = Student.new(age, @default_classroom, name, parent_permission: has_parent_permission)
```

Comment:
- [ ] there is an error while creating student and this is because you are passing four arguments in Student.new() instead of three kindly remove `@default_classroom` from this line of code
![image](https://user-images.githubusercontent.com/66097371/186890339-f7d4debb-50b6-4d6d-9b12-7583c6c2d1d6.png)